### ISSUE TITLE
Unable to connect outside of localhost

### ISSUE THREAD

**Issue by DrSeussFreak** [Jackett/Jackett#415]
On the machine running Jackett (Windows 10), I can access the app via localhost:port or ip:port, but I cannot access it from any other machine.  

Sonarr and Couchpotato have no issues connecting via IP, and I do have "External Access" checked.

**Comment by d2dyno** [comment]
By "Any other machine", do you mean inside or outside of your current network?

**Comment by DrSeussFreak** [comment]
Inside my network

**Comment by CatmanIX** [comment]
I'm also having the same issue. "External Access" is checked, ports are even forwarded, but i cannot connect from any other local network machine (or remote for that matter). I've even restarted the service a few times (I have not restarted my machine yet however).

Interestingly, connecting from the local system to the LAN IP (192.168.1.x) of that system works with "external access" enabled and doesn't when it's disabled (showing an "unauthorized host" error).

Anything else though doesn't even get an "unauthorized access" error, it just fails to reach the server at all. All tested devices are capable of connecting to other web sources on the server (qBT's web UI for example) however so it is definitely connectable.

**Comment by flightlevel** [comment]
Works here on iOS and PC.
Have you tried turning firewall off, antivrius off, security off?

**Comment by CatmanIX** [comment]
@flightlevel i have none of that enabled to start with, and like i said, other web apps are accessible with no issue.

**Comment by flightlevel** [comment]
@CatmanIX Good to hear its going

**Comment by DrSeussFreak** [comment]
Hi, I opened this for help, and I am still unable to access the app externally, can we please re-open?  I have confirmed that the windows firewall is not active.

**Comment by flightlevel** [comment]
Sure, reopne

**Comment by larserikjohansen** [comment]
I have the same problem on ubuntu 16x
when i do "mono /opt/jackett/JackettConsole.exe" it stars well, and at the end it says "09-06 21:22:52 Info Starting web server at http://127.0.0.1:9117/"

i am unable to connect it on the network, 
everything else with cp/sonarr/plex/ is working well.

**Comment by kamikaze2112** [comment]
I can't seem to get it working either.  I'm running it on windows, as far as I know the firewall isn't getting in the way as I run CouchPotato and SickRage in a similar fashion and didn't have to change any settings in the firewall for those.  I checked the setting in ServerConfig.json and it's set to true, so that's not the issue.  Not a big deal at the moment, since I'm only using it for ipTorrents at the moment.

**Comment by kaso17** [comment]
I can confirm that it seems like the external access feature is buggy.
I just tested it on windows 7:
* The External Access GUI option simply doesn't work. The config file must be edited manually as @NOsteyr already figured out.
* Jackett always listens on 0.0.0.0, even if AllowExternal is disabled. But if it's disabled you'll get a "Bad Gateway" 400 error (only tested access via IPv4 address). If it's set to false then jackett additionally listens on 127.0.0.1.
* If AllowExternal is set to true it stops listening on 127.0.0.1 and it can't be accessed locally anymore (Service Unavailable 503 error).

To DrSeussFreak and anyone else who can't access jackett remotely on windows: It's not enough to simply disable the windows firewall, you've to explicitly create a new rule allowing access to the jackett port.

**Comment by kamikaze2112** [comment]
I created a firewall rule that specifically allowed port 9117 TCP which has worked for external access.

The GUI option seems to work fine in my case, as the json file reads true for that option when it's checked in the GUI.  Also I can still access the GUI on the local machine with that option enabled.

**Comment by kaso17** [comment]
OK, GUI option is working fine for me too (Was a local issue).
I assume adding the firewall exception fixed it for you?

**Comment by kamikaze2112** [comment]
Yeah the firewall rule worked.  One thing I did notice is that if you change the port then you can't open the GUI from the tray icon. It seems to be hard coded for port 9117.  Not a big problem,  just something I noticed

**Comment by flightlevel** [comment]
@DrSeussFreak Did you ever get it going?

**Comment by DrSeussFreak** [comment]
Yes, it was Windows Firewall

**Comment by hitnrun30** [comment]
I thought I setup the Windows firewall correctly, but even so, I still cant get to Jackett from the outside. I open the ports on my router and added an inbound rule on my computer.
Can you either share the exact settings you did to setup the inbound rule or when the service is installed add the rule automatically.

**Comment by DrSeussFreak** [comment]
I created a new rule allowing specific ports on the inbound rules side (advanced settings area)

**Comment by hitnrun30** [comment]
Can you give me specifics, I'm not a noob, but I went into windows firewall and thought i did the same, but still it didnt work.  I opened it for port 9117 on that machine and the same on my router.

**Comment by DrSeussFreak** [comment]
- Go to "Control Panel"
- Window's Firewall
- Advanced Settings
- Inbound Rules
- "New Rule"
- "Port" --> Next
-  Specify Ports and whether it is TCP or UDP (this is TCP) --> Next
- "Allow the Connection" --> Next
- Leave all defaults --> Next
- Give it a name and description --> Finish

Test, you should be all set

**Comment by hitnrun30** [comment]
Awesome, I was setting the other port (remote) thank you.

**Comment by Eisa-AlAwadhi** [comment]
Also using Windows and I have the same issue, waiting patiently for a fix!

**Comment by flightlevel** [comment]
@Eisa01 An official fix would be changing the way Windows firewall works, I don't know how you expect us to do this? As a Windows user I would expect if I have the firewall turned off it to let everything through, but in reality this isn't what happens on Windows

**Comment by Eisa01** [comment]
@flightlevel 
Hmm, will it not be possible to add a rule for Jackett to bypass the Windows Firewall?
There's no need to turn off Windows firewall, just a rule that allows Jackett to bypass the firewall.

Upon installation or Jackett's first run, Jackett can ask for permission to bypass firewall similar to different programs.
Once we accept the rule then Jackett can bypass the firewall freely.

Right now, I've done this step manually by adding a rule for Jackett, so it might be possible to program it on Jackett's first run like torrent or many other programs?

**Comment by flightlevel** [comment]
Sure, I get what you mean. But, I don't think its as easy as first thought because Jackett generally runs as a service on Windows which doesn't have the same level of access as a GUI application. Will consider

**Comment by Watison** [comment]
I have this problem too. It works on my server's desktop using _127.0.0.1:9117_ or _localhost:9117_ but not _<fixedip>:9117_. All my other apps work fine remotely and I've tried an inbound rule on the remote PC, an outbound on the server and tried accessing from iOS.

**Comment by gosh7001** [comment]
I figured out that you have to do the same thing that you have to set the Firewall Rule for Port 9117 as suggested above for Windows 10. The ServerConfig.json file in C:\ProgramData\Jackett also must be modified --> "AllowExternal": false, to true   .
To modify in Windows, you must stop the jackett service, drag the ServerConfig.json file to desktop, make the change and then put it pack into ProgramData\Jackett directory.
Once this is done, start the service and you are now in business.

**Comment by goldenpipes** [comment]
i just did what @gosh7001 suggested but my ServerConfig.json is already set to true.

I have added a firewall rule as well. still no luck.  is the server being bound to 0.0.0.0?

**Comment by gosh7001** [comment]
what is in your hosts file (Windows/System32/drivers/etc/)? 
Make sure you have the following.
127.0.0.1        localhost
<yourIpHERE>     localhost

**Comment by goldenpipes** [comment]
checking now

**Comment by goldenpipes** [comment]
for some reason that entry was # out.

it now reads 127.0.0.1 localhost. without the #

**Comment by gosh7001** [comment]
is it working now?

**Comment by goldenpipes** [comment]
nope :( i know i have that port forwarded in my router too.

**Comment by gosh7001** [comment]
forget the router at this point ... just try to get it on the local machine ... something sounds funny about your windows install ... if you have not done so already, remove Jackett entirely, then reinstall with the "Run as Administrator" option.

**Comment by gosh7001** [comment]
If your firewall is on, turn it completely off too ... for Private, Local and Domain.

**Comment by goldenpipes** [comment]
pretty sure i did as administrator but i will try again.  I just rebooted the machine

**Comment by goldenpipes** [comment]
okay so i ran the uninstaller. and removed the folder from %programdata% and reinstalled as administrator.  I can access via 127.0.0.1 as i could before.  rechecked external access. restarted the service.  still nothing.  I also cannot even access it on another PC in my network.  I have Plex, ombi, plexpy, as well as radarr and sonarr running and accessible. not sure why this isnt working

**Comment by goldenpipes** [comment]
okay wtf. i turned off the firewall and it works.  but i have a firewall rule??

**Comment by goldenpipes** [comment]
any idea why the firewall rules get ignored?

**Comment by hitnrun30** [comment]
did you set it correctly, here is the setup for my firewall

![image](https://user-images.githubusercontent.com/23220612/31201182-b1dc7142-a92b-11e7-86da-22f652ae0d60.png)

**Comment by gosh7001** [comment]
delete the rule then add it back ... make sure the only thing you change is port 9117 on the INBOUND rule. --- make sure it is done for Local / Private / Domain ... I am thinking about what I had to do. Just a sec

**Comment by gosh7001** [comment]
Did you install as a service or just running on startup?

**Comment by hitnrun30** [comment]
Also see if your Antivirus firewall is stopping the application.  I have AVG and it constantly thinks its a suspicious program

**Comment by gosh7001** [comment]
and when you add it, the Enabled box is not checkmarked -- catch 22 ... you have to Enable it ... counter intuitive .

**Comment by goldenpipes** [comment]
installed as service. i will play around with the firewall  stuff :) thanks for the help guys i really appreciate it!

**Comment by acollien** [comment]
> I figured out that you have to do the same thing that you have to set the Firewall Rule for Port 9117 as suggested above for Windows 10. The ServerConfig.json file in C:\ProgramData\Jackett also must be modified --> "AllowExternal": false, to true .
> To modify in Windows, you must stop the jackett service, drag the ServerConfig.json file to desktop, make the change and then put it pack into ProgramData\Jackett directory.
> Once this is done, start the service and you are now in business.

Any idea where this file is nowadays - or where that variable is? I'm on Ubuntu 18.04 - there's no servercong file in the jacket folder anymore.

**Comment by jstahley** [comment]
I had this issue too on a Windows Server box. As stated above I shut the service off, changed the app json configuration file line to ""AllowExternal": true,". May  seem obvious but I just fell for this. Double clicking a string in notepad also highlights the trailing comma. If you do not replace the comma the configuration file revers back to ""AllowExternal": false," and will not work.

**Comment by bombz88** [comment]
> I had this issue too on a Windows Server box. As stated above I shut the service off, changed the app json configuration file line to ""AllowExternal": true,". May seem obvious but I just fell for this. Double clicking a string in notepad also highlights the trailing comma. If you do not replace the comma the configuration file revers back to ""AllowExternal": false," when restarting the service and will not accept remote connections.

Thank you, everyone above. Once I processed this step and edited the json file I was able to connect to the Web UI from within the internal network.
Much apprecaited!

**Comment by mike9823** [comment]
> did you set it correctly, here is the setup for my firewall
> 
> ![image](https://user-images.githubusercontent.com/23220612/31201182-b1dc7142-a92b-11e7-86da-22f652ae0d60.png)

With the ServerConfig file edited as well as changing the rule to TCP 9117, it's working.

**Comment by Asgardsurfer** [comment]
**I had this issue too on a Windows Server box. As stated above I shut the service off, changed the app json configuration file line to ""AllowExternal": true,". May seem obvious but I just fell for this. Double clicking a string in notepad also highlights the trailing comma. If you do not replace the comma the configuration file revers back to ""AllowExternal": false," when restarting the service and will not accept remote connections.**
------------------------------------------------------------------------------------------------------------------------------
Just experienced the same issue, unable to connect to Jackett on my iPhone or iPad  through the LAN. Was working perfectly before. Works perfectly on "localhost"
I checked firewall, VPN, Adguard etc...nothing had changed...
MacOS 10.14.6
Here is how I modified the ServerConfig.json
Used the app "find any File" and searched ServerConfig.json and made hidden files visible.
Once located, I opened the file with BBEdit. 
Then launchctl unload ~/Library/LaunchAgents/org.user.Jackett.plist in Terminal
Modified in BBEdit from "false" to "true", save in BBEdit
Then launchctl load ~/Library/LaunchAgents/org.user.Jackett.plist in Terminal

WORKS!

Thanks to all that posted in this thread.
